radix-engine/src/engine/kernel.rs

```diff
@@ -1615,3 +1375,261 @@ where
+                    RuntimeError::InterpreterError(
+                        InterpreterError::InvalidScryptoMethodInvocation(
+                            method_ident.clone(),
+                            ScryptoFnResolvingError::MethodNotFound,
+                        ),
+                    ),
+                )?;
+                if fn_abi.mutability.is_none() {
+                    return Err(RuntimeError::InterpreterError(
+                        InterpreterError::InvalidScryptoMethodInvocation(
+                            method_ident.clone(),
+                            ScryptoFnResolvingError::MethodNotFound,
+                        ),
+                    ));
+                }
+
+                // Check input against the ABI
+                if !fn_abi.input.matches(&args.dom) {
+                    return Err(RuntimeError::InterpreterError(
+                        InterpreterError::InvalidScryptoMethodInvocation(
+                            method_ident.clone(),
+                            ScryptoFnResolvingError::InvalidInput,
+                        ),
+                    ));
+                }
+
+                // Emit event
+                for m in &mut self.modules {
+                    m.on_wasm_instantiation(
+                        &self.current_frame,
+                        &mut self.heap,
+                        &mut self.track,
+                        package.code(),
+                    )
+                    .map_err(RuntimeError::ModuleError)?;
+                }
+
+                (
+                    self.scrypto_interpreter
+                        .create_executor(&package.code, invocation.args().clone()),
+                    REActor::Method(
+                        ResolvedMethod::Scrypto {
+                            package_address: component_info.package_address,
+                            package_id: package_node_id.into(),
+                            blueprint_name: component_info.blueprint_name,
+                            ident: method_ident.method_name.clone(),
+                            export_name: fn_abi.export_name.clone(),
+                            return_type: fn_abi.output.clone(),
+                        },
+                        resolved_receiver,
+                    ),
+                )
+            }
+        };
+
+        for global_address in invocation.args().global_references() {
+            node_refs_to_copy.insert(RENodeId::Global(global_address));
+        }
+
+        node_refs_to_copy.insert(RENodeId::Global(GlobalAddress::Resource(RADIX_TOKEN)));
+        node_refs_to_copy.insert(RENodeId::Global(GlobalAddress::System(EPOCH_MANAGER)));
+        node_refs_to_copy.insert(RENodeId::Global(GlobalAddress::Resource(
+            ECDSA_SECP256K1_TOKEN,
+        )));
+        node_refs_to_copy.insert(RENodeId::Global(GlobalAddress::Resource(
+            EDDSA_ED25519_TOKEN,
+        )));
```

Comment:
Why do we still need this? To allow scrypto direct access to these entities without ref passing?

Reply:
Yup we have a couple of examples which still do this.